scripts/irods/test/resource_suite.py

```diff
@@ -513,68 +513,80 @@ def test_local_iput_onto_specific_resource(self):
+        restartfile = os.path.join(self.admin.local_session_dir, 'collectionrestartfile')
+        iputcmd = f'iput -X {restartfile} -r {local_dir_path} {collection_path}'
+
+        os.makedirs(local_dir_path)
         for datafilename in datafiles:
             print("-------------------")
             print("creating " + datafilename + "...")
-            localpath = datadir + "/" + datafilename
+            localpath = local_dir_path + "/" + datafilename
             lib.make_file(localpath, 2**20)
-        restartfile = "collectionrestartfile"
-        # assertions
-        iputcmd = "iput -X " + restartfile + " -r " + datadir
-        if os.path.exists(restartfile):
-            os.unlink(restartfile)
-        self.admin.interrupt_icommand(iputcmd, restartfile, 10)  # once restartfile reaches 10 bytes
-        assert os.path.exists(restartfile), restartfile + " should now exist, but did not"
+
+        # once restartfile reaches 10 bytes
+        self.admin.interrupt_icommand(iputcmd, restartfile, 10)
+        # Wait for all interrupted threads to exit - the files are large enough to trigger parallel transfer and some
+        # suites that run this test may be running some replication, sync-to-archive, etc.
+        # TODO: #6879 - This sleep will not be necessary if iput disconnects on SIGTERM.
+        time.sleep(10)
+
+        self.assertTrue(os.path.exists(restartfile), msg=f'{restartfile} should now exist, but did not')
+
         print("  restartfile [" + restartfile + "] contents --> [")
         with open(restartfile, 'r') as f:
             for line in f:
                 print(line)
         print("]")
-        self.admin.assert_icommand("ils -L " + datadir, 'STDOUT_SINGLELINE', datadir)  # just to show contents
-        self.admin.assert_icommand(iputcmd, 'STDOUT_SINGLELINE', "File last completed")  # confirm the restart
-        for datafilename in datafiles:
-            self.admin.assert_icommand("ils -L " + datadir, 'STDOUT_SINGLELINE', datafilename)  # should be listed
-        # local cleanup
-        lib.execute_command(['rm', '-rf', datadir])
-        if os.path.exists(restartfile):
-            os.unlink(restartfile)
 
-    @unittest.skipIf(True, 'Enable once race conditions fixed, see #2634')
+        _,out,_ = self.admin.assert_icommand(['ils', '-L', collection_path], 'STDOUT', collection_path)
+        some_files_missing_from_collection = False
+        for datafile in datafiles:
+            if datafile not in out:
+                some_files_missing_from_collection = True
+
+        # Some of these should have failed to upload.
+        self.assertTrue(some_files_missing_from_collection)
+
+        # Confirm the restart occurs and all of the files are present as data objects in the collection.
+        self.admin.assert_icommand(iputcmd, 'STDOUT', "File last completed")
+        _,out,_ = self.admin.assert_icommand(['ils', '-L', collection_path], 'STDOUT', collection_path)
```

Comment:
Can `desired_rc` here too for bonus check.

Reply:
Done